prow/cmd/sinker/main.go

```diff
@@ -124,15 +135,44 @@ func main() {
 		config:        cfg,
 	}
 
-	// Clean now and regularly from now on.
-	for {
-		start := time.Now()
-		c.clean()
-		logrus.Infof("Sync time: %v", time.Since(start))
-		if o.runOnce {
-			break
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		// Clean now and regularly from now on.
+		for {
+			start := time.Now()
+			c.clean()
+			logrus.Infof("Sync time: %v", time.Since(start))
+			if o.runOnce {
+				break
+			}
+			time.Sleep(cfg().Sinker.ResyncPeriod)
 		}
-		time.Sleep(cfg().Sinker.ResyncPeriod)
+	}()
+
+	if !o.runOnce {
+		// Return 200 on / for health checks.
+		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {})
+		http.Handle("/metrics", promhttp.Handler())
+
+		httpServer := &http.Server{Addr: ":" + strconv.Itoa(o.port)}
+
+		// Shutdown gracefully on SIGTERM or SIGINT
+		sig := make(chan os.Signal, 1)
```

Comment:
We also don't care about graceful shutdown for the metrics server (unlike `hook` does for the `hook` server) so we can get away with:

```go
metricsMux := http.NewServeMux()
metricsMux.Handle("/metrics", promhttp.Handler())
go func() {
    logrus.WithError(http.LitstenAndServe(":9090", metricsMux)).Fatal("ListenAndServer returned while serving metrics.")
}
```

Reply:
Done.